### PR TITLE
WIP Add mysql

### PR DESCRIPTION
**Pull request by NiR-**
TODO:

* [ ] Need to test mysql backup/restore tools

### CODE REVIEW

File: mysql/backup [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+set -o nounset
+set -o pipefail
+set -o errexit
+set -o errtrace
+[ -n "${TRACE:-}" ] && set -o xtrace
+
+backup_file=${PGDATABASE}-$(date +%d%m%Y%H%M%S).sql.gz
```
Comment (nicolasmure) [inline mysql/backup]: `PGDATABASE` => `DATABASE` ? it's not about PG here :)

File: mysql/backup [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+backup_path=/tmp/${backup_file}
+
+mysqldump ${@} | gzip > ${backup_path}
+
+gcloud auth activate-service-account --key-file=${GCLOUD_SERVICE_ACCOUNT_KEY_FILE}
+gsutil -h "x-goog-meta-db-backups-to-gcs-version:mysql-${IMAGE_VERSION}" \
+    cp \
+    -a project-private \
+    ${backup_path} gs://${BUCKET_NAME}/${backup_file}
```
Comment (nicolasmure) [inline mysql/backup]: can't this be streamed directly from the `mysqldump` output instead of relying on a tmp file ?

File: mysql/restore [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+backup_file="${@: -1}"
+backup_path="/tmp/${backup_file}"
+
+unset 'arr[${#arr[@]}-1]'
+
+gcloud auth activate-service-account --key-file=${GCLOUD_SERVICE_ACCOUNT_KEY_FILE}
+gsutil cp gs://${BUCKET_NAME}/${backup_file} ${backup_path}
+
+gunzip ${backup_path} | mysql "${@}"
```
Comment (nicolasmure) [inline mysql/restore]: same here : what about directly loading the dumpfile instead of writing it on the FS ?

File: mysql/backup (hunk 2 above)
Comment (nicolasmure) [reply mysql/backup]: You'd avoid to take twice as much space as needed IMHO ;)

**Comment by NiR-** [comment]
@nicolasmure I rewrote the backup/restore scripts for MySQL to better follow our internal bash guidelines and to add support for some env vars.

File: docs/mysql.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+With Docker:
+
+```bash
+docker run --rm -it \
+  --env GCLOUD_SERVICE_ACCOUNT_KEY_FILE=/gcloud-sa.json
+  --env BUCKET_NAME=db-backup
+  --volume ./gcloud-sa.json:/gcloud-sa.json:ro
+  akerouanton/db-backups-to-gcs:mysql-restore-0.2
+  my_db-20181018131600.tar
```
Comment (nicolasmure) [inline docs/mysql.md]: isn't it a `.sql.gz` file instead ?

File: docs/postgres.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+docker run --rm -it \
+  --env PGHOST=db
+  --env PGDATABASE=my_db
+  --env PGUSER=admin
+  --env PGPASSWORD=my_password
+  --env GCLOUD_SERVICE_ACCOUNT_KEY_FILE=/gcloud-sa.json
+  --env BUCKET_NAME=db-backup
+  --volume ./gcloud-sa.json:/gcloud-sa.json:ro
+  akerouanton/db-backups-to-gcs:postgres-restore-0.2
+  my_db-20181018131600.tar
```
Comment (nicolasmure) [inline docs/postgres.md]: same here ?

File: mysql/backup [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    if [ -z "${MYSQL_DATABASE:-}" ]; then
+        printerr "Please define MYSQL_DATABASE env var."
+    fi
+
+    if [ -n "${MYSQL_PASSWORD:-}" ]; then
+        export MYSQL_PWD="${MYSQL_PASSWORD}"
+    fi
+}
+
+check_env_vars
```
Comment (nicolasmure) [inline mysql/backup]: what about wrapping these calls in a `main` function ?

File: mysql/restore [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        cp gs://${BUCKET_NAME}/${backup_file} - \
+    | gunzip \
+    | mysql \
+        --host "${MYSQL_HOST:-}" \
+        --user "${MYSQL_USER:-}" \
+        ${mysql_options} \
+        ${MYSQL_DATABASE}
+}
+
+check_env_vars() {
```
Comment (nicolasmure) [inline mysql/restore]: maybe you can put this fn in a dedicated file and call in from this `restore` and `backup` script ?

File: mysql/restore [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+        export MYSQL_PWD="${MYSQL_PASSWORD}"
+    fi
+}
+
+if [ "${#}" -eq 0 ]; then
+    echo "Usage: ${@} [mysql-options] <backup-filename>" >&2
+    exit 1
+fi
+
+check_env_vars
```
Comment (nicolasmure) [inline mysql/restore]: same here : use a `main` fn ?

File: mysql/backup [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+    local mysql_options="${@}"
+    local backup_file=${MYSQL_DATABASE}-$(date +%d%m%Y%H%M%S).sql.gz
+
+    mysqldump \
+        --user "${MYSQL_USER}" \
+        --host "${MYSQL_HOST}" \
+        ${mysql_options} \
+        ${MYSQL_DATABASE} \
+    | gzip \
+    | gsutil \
```
Comment (nicolasmure) [inline mysql/backup]: :+1: pipes :sunglasses:

File: mysql/restore (hunk 7 above)
Comment (NiR-) [reply mysql/restore]: I thought about that but imho better KISS as both scripts won't even be co-located on the same image (there's one Docker image per script).

**Pull request closed without merging**